salt/beacons/salt_monitor.py

```diff
@@ -0,0 +1,118 @@
+"""
+A beacon to execute salt execution module functions. This beacon will fire only if the return data is "truthy".
+The function return, funtion name and args and/or kwargs, will be passed as data in the event.
+
+The configuration can accept a list of salt functions to execute every interval.
+Make sure to allot enough time via 'interval' key to allow all salt functions to execute.
+The salt functions will be executed sequentially.
+
+The elements in list of functions can be either a simple string (with no arguments) or a dictionary with a single
+key being the salt execution module and sub keys indicating args and / or kwargs.
+
+See example config below.
+
+.. code-block:: yaml
+
+    beacons:
+      salt_monitor:
+        - salt_fun:
+        - slsutil.renderer:
+            args:
+              - salt://states/apache.sls
+            kwargs:
+              - default_renderer: jinja
+        - test.ping
+        - interval: 3600 # seconds
+"""
+
+
+def _parse_args(args_kwargs_dict):
+    args = args_kwargs_dict.get("args", [])
+    kwargs = args_kwargs_dict.get("kwargs", {})
+    if kwargs:
+        _kwargs = {}
+        list(map(_kwargs.update, kwargs))
+        kwargs = _kwargs
+
+    return args, kwargs
+
+
+def validate(config):
+    _config = {}
+    list(map(_config.update, config))
```

Comment:
`config` is an iterable due to a somewhat unfortunate convention used in the config files: the configuration is supplied as a list of single key dictionaries. So through the use of these couple of lines we convert that to a simple multi-key dictionary. The `list()` function is used because map is a generator in recent versions of python. The lines of code I used here can be found in other beacon modules, for example [inotify](https://github.com/saltstack/salt/blob/master/salt/beacons/inotify.py#L112-L113).

Reply:
Thanks for the clarification.